File: hacker-rank-interview-preparation-kit/Ruby/warm-up-challenges/jumping_on_the_clouds.rb
=begin
Tae H0 K1m
  3. Jumping on the Clouds

  Determine the minimum number of jumps it will take Emma to jump from her
  starting postion to the last cloud. It is always possible to win the game.

  An array of clouds numbered 0 if they are safe or 1 if they must be avoided.
  Emma can jump on any cloud having a number that is equal to the number of the
  current cloud plus 1 or 2.

  It should return the minimum number of jumps required, as an integer.

  ** 2 <= n <= 100  ; the total number of clouds
=end

def jumpingOnClouds(c)
  jumps = 0
  index = 0

  until index == c.length - 1
    if !c[index + 2].nil? && c[index + 2] != 1
      index = index + 2
    else
      index = index + 1
    end
    jumps += 1
  end


  return jumps
end

puts jumpingOnClouds([0,0,1,0,0,1,0])
# => 4

puts jumpingOnClouds([0,0,0,0,1,0])
# => 3

File: hacker-rank-interview-preparation-kit/Python/warm-up-challenges/jumping_on_the_clouds.py
""" #3 Jumping on the Clouds
    Tae H0 K1m

    problem: You wants to jump from the first element to the last element in
    the given array (an array of binary integers). You need to avoid 1. Output
    is the minimum number of jumps required.

    input: an array of binary integers
    output: the minimum number of jumps required

    approach:
        1. iterate the array
        2. check if the element in the current index plus 2 is 0, then update
        index plus 2, otherwise plus 1,
        3. increment the number of jumps
"""

def jumpingOnClouds(c):
    jumps = 0
    i = 0

    while i < len(c) - 1:
      if i+2 < len(c) and c[i+2] == 0:
        i += 2
      else:
        i += 1

      jumps += 1

    return jumps


print(jumpingOnClouds([0,0,1,0,0,1,0]))
# => 4

print(jumpingOnClouds([0,0,0,0,1,0]))
# => 3

print(jumpingOnClouds([0,0]))
# => 1
